custom_components/smartir/light.py

```diff
@@ -150,155 +127,189 @@ def extra_state_attributes(self):
-        await self.send_remote_command(remote_cmd, count)
-
-    async def send_remote_command(self, remote_cmd, count=1):
+    async def send_command(self, state, brightness, color_temp):
         async with self._temp_lock:
-            self._on_by_remote = False
+            if self._power_sensor and self._state != state:
+                self._async_power_sensor_check_schedule(state)
+
             try:
-                for _ in range(count):
-                    await self._controller.send(remote_cmd)
-                    await asyncio.sleep(self._delay)
+                if state == STATE_OFF:
+                    if "off" in self._commands.keys() and isinstance(
+                        self._commands["off"], str
+                    ):
+                        if (
+                            "on" in self._commands.keys()
+                            and isinstance(self._commands["on"], str)
+                            and self._commands["on"] == self._commands["off"]
+                            and self._state == STATE_OFF
+                        ):
+                            # prevent to resend 'off' command if same as 'on' and device is already off
+                            _LOGGER.debug(
+                                "As 'on' and 'off' commands are identical and device is already in requested '%s' state, skipping sending '%s' command",
+                                self._state,
+                                "off",
+                            )
+                        else:
+                            _LOGGER.debug("Found 'off' operation mode command.")
+                            await self._controller.send(self._commands["off"])
+                            await asyncio.sleep(self._delay)
+                    else:
+                        _LOGGER.error("Missing device IR code for 'off' mode.")
+                        return
+                else:
+                    if "on" in self._commands.keys() and isinstance(
+                        self._commands["on"], str
+                    ):
+                        if (
+                            "off" in self._commands.keys()
+                            and isinstance(self._commands["off"], str)
+                            and self._commands["off"] == self._commands["on"]
+                            and self._state == STATE_ON
+                        ):
+                            # prevent to resend 'on' command if same as 'off' and device is already on
+                            _LOGGER.debug(
+                                "As 'on' and 'off' commands are identical and device is already in requested '%s' state, skipping sending '%s' command",
+                                self._state,
+                                "on",
+                            )
+                        else:
+                            # if on code is not present, the on bit can be still set later in the all operation/fan codes"""
```

Comment:
This line ends with """.

The highlighting in github does not treat the rest of the file as a comment, so maybe this is OK because it is in a comment. But in my editor it was causing highlighting to display everything below as a comment, so I think it is better to remove it even if it is harmless.

Reply:
Agree, this was strange comment, it seems to be in all classes probably even from the original code - removed everywhere.